addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -379,14 +379,42 @@ func (h *Helper) HandleExistingClusterBootstrap(clusterBootstrap *runtanzuv1alph
 			packagesToBeCloned = append(packagesToBeCloned, nonEmptyInlinePackages...)
 			packages = packagesToBeCloned
 		}
+	} else {
+		// packages with values from inline need to be cloned for two cases:
+		//
+		// 1. When no Clusterbootstrap is given by user and clusterbootstraptemplate
+		//    is cloned as clusterbootstrap but resolved tkr is not set - In this
+		//   case the providers are already cloned so providers have to be cloned again
+		//
+		// 2. When full clusterbootstrap without tkr annotation is given by user.
+		//    In this case we only want to handle inline packages for creating
+		//    secret. The providers are expected to be present in the
+		//   clusternamespace, created by user before clusterbootstrap is created.
+		var nonEmptyInlinePackages []*runtanzuv1alpha3.ClusterBootstrapPackage
+		for _, cbPackage := range packages {
+			if cbPackage != nil && cbPackage.ValuesFrom != nil && cbPackage.ValuesFrom.Inline != nil {
+				nonEmptyInlinePackages = append(nonEmptyInlinePackages, cbPackage)
+			}
+		}
+		var packagesToBeCloned []*runtanzuv1alpha3.ClusterBootstrapPackage
+		packagesToBeCloned = append(packagesToBeCloned, nonEmptyInlinePackages...)
+		packages = packagesToBeCloned
 	}
 
-	secrets, providers, err := h.CloneReferencedObjectsFromCBPackages(cluster, packages, systemNamespace)
+	secrets, _, err := h.CloneReferencedObjectsFromCBPackages(cluster, packages, systemNamespace)
 	if err != nil {
 		h.Logger.Error(err, "unable to clone secrets or providers")
 		return nil, err
 	}
 
+	providers, err := h.getListOfProvidersFromRef(clusterBootstrap)
```

Comment:
I thought about the case where in the prior release what if any object didnt have owner reference ? Then the new config controller wouldnt work. i.e. I think we are breaking backwards compatibility with this change.

To be safe I think we need to set ownerreference for all providers outside this function in the reconcileNormal function as well. i.e. In any case set the ownerreference for all providers.

Reply:
added logic to add owner reference to any existing object listed under clusterbootstrap on reconcilenormal of the clusterbootstrap controller